grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/gradle/variant/Variant.kt

```diff
@@ -66,18 +67,23 @@ enum class DefaultVariants(val variantName: String) {
     },
     AndroidTest("androidTest") {
         override fun toString() = variantName
+    },
+    Lint("lint") {
+        override fun toString() = variantName
     }
 }
 
 val DEFAULT_VARIANT = DefaultVariants.Default.toString()
 val TEST_VARIANT = DefaultVariants.Test.toString()
 val ANDROID_TEST_VARIANT = DefaultVariants.AndroidTest.toString()
+val LINT_VARIANT = DefaultVariants.Lint.toString()
 
 enum class VariantType {
     AndroidBuild,
     AndroidTest,
     Test,
     JvmBuild,
+    Lint,
```

Comment:
Thought about this again, I think we can avoid creating a new variant for this to simplify stuff later on as explained below:

1. Goal of the variant API is to have source of truth for project variant and as well as abstract all configurations behind Variant instance. Adding new variant seems wasteful since Lint is not actually a variant, but just a separate classpath i.e. `lintChecks` is simply a classpath similar to `kapt` or `compilerPlugins`. That means we can consider adding a new field to [Variant](https://github.com/grab/grazel/blob/d4df4ac8f33f16be7e77f0a69d0254ebaf2ee74e/grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/gradle/variant/Variant.kt#L49-L57) like `lintConfiguration`.
2. Doing 1, also means we don't need to change rest of the code, this parsed `lintChecks` configuration will be available in `MatchedVariant` by default and can use that instead of reading `lintChecks` manually again like [this](https://github.com/grab/grazel/blob/1cfeb83dc863502931c564dcb184fcb151f91e20/grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/android/LintExtractor.kt#L28).
3. Existing configuration parsing is covered by tests, so tests need to be updated to handle `lintChecks`

Reply:
As discussed internally, I think let's keep the current structure as-is. I was wondering if lintChecks will be in a different classpath i.e `@lint_maven` and I just learnt it already does due to how VariantAPI is structured. So we can go ahead with the current change since this is relatively less code than manually updating the logic in resolve dependencies/workspace dependencies flow.